### PR TITLE
C driver handle untethered subscritions correctly

### PR DESCRIPTION
**Pull request by mikeb01**
- Check for the number of "working" tethered positions instead of the length of the array.
- Update IPC Publication, Network Publication and Publication Image to handle these changes.

### CODE REVIEW

File: aeron-driver/src/main/c/aeron_driver_conductor.c [hunk 1]
```diff
@@ -3225,6 +3230,48 @@ int aeron_driver_conductor_link_subscribable(
...
+    size_t count = 0;
+
+    for (int lastIndex = subscribable->length - 1, i = lastIndex; i >= 0; i--)
+    {
+        if (AERON_SUBSCRIPTION_TETHER_RESTING != subscribable->array[i].state)
+        {
+            count++;
+        }
+    }
+
```
Comment (reissGRVS) [inline aeron-driver/src/main/c/aeron_driver_conductor.c:3266]: Is this just `subscribable->length - subscribable->num_resting`?

File: aeron-driver/src/main/c/aeron_publication_image.c [hunk 2]
```diff
@@ -386,6 +385,11 @@ void aeron_publication_image_track_rebuild(aeron_publication_image_t *image, int
             }
         }
 
+        if (INT64_MAX == min_sub_pos)
+        {
+            return;
+        }
```
Comment (reissGRVS) [inline aeron-driver/src/main/c/aeron_publication_image.c:391]: I guess it should be impossible to hit this case due to the aeron_driver_subscribable_has_working_positions check?

File: aeron-driver/src/main/c/aeron_driver_conductor.c (hunk 1 above)
Comment (mikeb01) [reply aeron-driver/src/main/c/aeron_driver_conductor.c]: Yes, missed that optimisation.

File: aeron-driver/src/main/c/aeron_publication_image.c (hunk 2 above)
Comment (mikeb01) [reply aeron-driver/src/main/c/aeron_publication_image.c:391]: Yes, it is a belt & braces.

File: aeron-driver/src/main/c/aeron_publication_image.c [hunk 3]
```diff
@@ -904,7 +897,7 @@ void aeron_publication_image_on_time_event(
             bool is_end_of_stream;
             AERON_GET_VOLATILE(is_end_of_stream, image->is_end_of_stream);
 
-            if (0 == image->conductor_fields.subscribable.length ||
+            if (!aeron_driver_subscribable_has_working_positions(&image->conductor_fields.subscribable) ||
```
[the remark this answers is not in the document]
Comment (mikeb01) [reply aeron-driver/src/main/c/aeron_publication_image.c:900]: No, at that point no new positions should be added to the image and the image should be removed.  This behaviour is consistent with the Java driver.

**Review by reissGRVS — APPROVED** [review]

**Pull request merged**